---
title: "RNA Seq Analysis: Neurotoxicity Detection Project"
output: html_notebook
---
#Continuation of Exploratory Data Analysis
Here I will continue the exploratory data analysis using R.
```{r}
#Load dataframes manipulated from Jupyter Notebook
avg_genes=read.csv('/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/avg_genes.csv')
avg_genes_nontoxic=read.csv('/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/avg_genes_nontoxic.csv')
avg_genes_toxic=read.csv('/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/avg_genes_toxic.csv')
Day_16_RNA_seq=read.csv('/Users/danielaquijano/Documents/GitHub/Machine-Learning-Course-Projects/Final_Project_Neurotoxicity_Prediction/Final_Project_Files/Day_16_RNA_seq.csv')
```
```{r}

```

```{r}
avg_genes
```
```{r}
avg_genes$X <- NULL 
rownames(avg_genes)<-Day_16_RNA_seq$Gene.ID 
```
```{r}
avg_genes
```

```{r}
#Install Relevant Packages
install.packages("devtools")
```
```{r}
install.packages('dendextend')
install.packages('dendextendRcpp')
```
```{r}
BiocManager::install("umap", update = FALSE)
```

```{r}
#Load Relevant Libraries

library(pheatmap)#plotting heatmaps
library(grid)#plotting
library(gplots) #plotting
library(dendextend) #For dendogram visualization
library(DESeq2)#RNA Seq Analysis
library(umap)
library(dplyr)#For manipulation of dataframes
library(tidyverse) #PCA Analysis
```

```{r}
avg_genes_correlation<-cor(avg_genes,method = "spearman")
```

```{r}
pheatmap(avg_genes_correlation,color=rainbow(50),fontsize_row=4, fontsize_col=4,border_color=NA)
```
```{r}
heatmap.2(as.matrix(avg_genes_correlation),
          main = "Heatmap for the Average gene count on Day 16 and 21 after chemical exposure",cexRow=0.5
         )
```
```{r}

```

## Dend-extend package to visualize relationships between samples 
```{r}

iris_species <- rev(levels(iris[,5]))



# order it the closest we can to the order of the observations:
dend <- rotate(dend, 1:150)

# Color the branches based on the clusters:
dend <- color_branches(dend, k=3) #, groupLabels=iris_species)

# Manually match the labels, as much as possible, to the real classification of the flowers:
labels_colors(dend) <-
   rainbow_hcl(3)[sort_levels_values(
      as.numeric(iris[,5])[order.dendrogram(dend)]
   )]

# We shall add the flower type to the labels:
labels(dend) <- paste(as.character(iris[,5])[order.dendrogram(dend)],
                           "(",labels(dend),")", 
                           sep = "")
# We hang the dendrogram a bit:
dend <- hang.dendrogram(dend,hang_height=0.1)
# reduce the size of the labels:
# dend <- assign_values_to_leaves_nodePar(dend, 0.5, "lab.cex")
dend <- set(dend, "labels_cex", 0.5)
# And plot:
par(mar = c(3,3,3,7))
plot(dend, 
     main = "Clustered Iris data set
     (the labels give the true flower species)", 
     horiz =  TRUE,  nodePar = list(cex = .007))
legend("topleft", legend = iris_species, fill = rainbow_hcl(3))
```


#Differential Gene Expression Analysis using DESeq2
Here I will be using the DESeq2 package from bioconductor in order to conduct differential gene expression analysis. I will begin the analysis with some additional exploratory data asnalysis including Principal Component Analysis (PCA).


```{r}
expgroup<-data_frame()
condition_toxic<-rep('Toxic',68,)
condition_nontoxic<-rep('Nontoxic',72)

```
```{r}
condition<-c(condition_nontoxic,condition_toxic)
samples<-colnames(avg_genes)
expgroup<-cbind(condition)
rownames(expgroup)<-samples
```
```{r}
expgroup #The purpose of the expgroup matrix is to label each sample with the condition it was subjected to. In this case its the exposure of either a toxic or non toxic chemical
```

UMAP steps: https://alexslemonade.github.io/refinebio-examples/03-rnaseq/dimension-reduction_rnaseq_02_umap.html 

## Uniform Manifold Approximation and Projection in R (UMAP)
```{r}

```


##Principal Component Analysis (PCA)
```{r}
#Principal component analysis for data before differential gene expression analysis
#We expect data to cluster depending on whether the neural construct was exposed to a toxic chemical or not.

pca_matrix_avg_genes <- as.matrix(t(avg_genes)) #Transpose avg_genes to flip columns with rows. 
```
```{r}
sample_pca <- prcomp(pca_matrix_avg_genes)
```
```{r}

as_tibble(pca_matrix_avg_genes)# Convert matrix to tibble (a tibble is a tidyverse version of a dataframe)
```
```{r}

```



## T-distributed stochastic neighbor embedding (TSNE)
```{r}

```

